Extension/src/LanguageServer/extension.ts

```diff
@@ -433,6 +436,73 @@ export function registerCommands(enabled: boolean): void {
+    // ---------------- Wrappers -------------
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.ConfigurationSelectUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("ConfigurationSelect");
+            onSelectConfiguration();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.RescanWorkspaceUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("RescanWorkspace");
+            onRescanWorkspace();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.ShowIdleCodeAnalysisCommandsUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("ShowIdleCodeAnalysisCommands");
+            onShowIdleCodeAnalysisCommands();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.ShowActiveCodeAnalysisCommandsUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("ShowActiveCodeAnalysisCommands");
+            onShowActiveCodeAnalysisCommands();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.PauseParsingUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("ParsingCommands");
+            onPauseParsing();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.ResumeParsingUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("ParsingCommands");
+            onResumeParsing();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.CheckForCompilerUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("CheckForCompiler");
+            onCheckForCompiler();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.RestartIntelliSenseForFileUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("RestartIntelliSenseForFile");
+            onRestartIntelliSenseForFile();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.ShowReferencesProgressUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("ShowReferencesProgress");
+            onShowReferencesProgress();
+        }
+        : onDisabledCommand));
+    commandDisposables.push(vscode.commands.registerCommand('C_Cpp.ShowParsingCommandsUI_Telemetry', enabled ?
+        () => {
+            logForUIExperiment("ParsingCommands");
+            onShowParsingCommands();
+        }
+        : onDisabledCommand));
+    // ----------------------------------------
+}
+
+async function logForUIExperiment(command: string): Promise<void> {
+    const whichUI: string = (await getUI()).whichUI ? "New UI" : "Old UI";
```

Comment:
Just a convention, we don't usually put spaces in the event names.  And something like this would be better logged as a boolean in the payload instead of concatenating it into the event name.

Consider this:
```ts
  const newUI = (await getUI()).whichUI.toString();
  telemetry.logLanguageServerEvent(`experiment${command}`, { newUI });
```